packages/rpc-core/src/rpc-methods/__tests__/get-program-accounts-test.ts

```diff
@@ -0,0 +1,1214 @@
+                });
+            });
+        });
+    });
+
+    describe('when called with a `minContextSlot` higher than the highest slot available', () => {
+        it('throws an error', async () => {
+            expect.assertions(1);
+            const program =
+                'DXngmJfjurhnAwbMPgpUGPH6qNvetCKRJ6PiD4ag4PTj' as Base58EncodedAddress<'DXngmJfjurhnAwbMPgpUGPH6qNvetCKRJ6PiD4ag4PTj'>;
+            const sendPromise = rpc
+                .getProgramAccounts(program, {
+                    minContextSlot: 2n ** 63n - 1n, // u64:MAX; safe bet it'll be too high.
+                })
+                .send();
+            await expect(sendPromise).rejects.toMatchObject({
+                code: -32016 satisfies (typeof SolanaJsonRpcErrorCode)['JSON_RPC_SERVER_ERROR_MIN_CONTEXT_SLOT_NOT_REACHED'],
+                message: expect.any(String),
+                name: 'SolanaJsonRpcError',
+            });
+        });
+    });
+
+    describe('when called with a program with no accounts', () => {
+        it('returns an empty list', async () => {
+            expect.assertions(1);
+            // randomly generated
+            const program =
+                'FZp6uJxwZbyZUCSF3Hyv494Su7w2MJEtNdYuc4RQHa2Z' as Base58EncodedAddress<'FZp6uJxwZbyZUCSF3Hyv494Su7w2MJEtNdYuc4RQHa2Z'>;
+            const accountInfoPromise = rpc.getProgramAccounts(program).send();
+            await expect(accountInfoPromise).resolves.toStrictEqual([]);
+        });
+    });
+
+    describe('when called with a program with multiple accounts', () => {
+        it('returns account info for all accounts', async () => {
+            expect.assertions(3);
+            // See scripts/fixtures/gpa2-1.json, scripts/fixtures/gpa2-2.json,
+            const program =
+                'AmtpVzo6H6qQCP9dH9wfu5hfa8kKaAFpTJ4aamPYR6V6' as Base58EncodedAddress<'AmtpVzo6H6qQCP9dH9wfu5hfa8kKaAFpTJ4aamPYR6V6'>;
+
+            const accountInfo = await rpc.getProgramAccounts(program).send();
+
+            expect(accountInfo).toHaveLength(2);
+            expect(accountInfo[0]).toMatchObject({
+                account: expect.objectContaining({
+                    data: expect.any(String),
+                    executable: expect.any(Boolean),
+                    lamports: expect.any(BigInt),
+                    owner: expect.any(String),
+                    rentEpoch: expect.any(BigInt),
+                    space: expect.any(BigInt),
+                }),
+                pubkey: expect.any(String),
+            });
+            expect(accountInfo[1]).toMatchObject({
+                account: expect.objectContaining({
+                    data: expect.any(String),
+                    executable: expect.any(Boolean),
+                    lamports: expect.any(BigInt),
+                    owner: expect.any(String),
+                    rentEpoch: expect.any(BigInt),
+                    space: expect.any(BigInt),
+                }),
```

Comment:
Should we actually assert on the values of the accounts, since they come from fixtures?

Reply:
We definitely could, we just have a lot of tests already that don't

Reply:
I sort of think we should. It's an easy lay-up for completing some of these tests and asserting proper functionality. Happy to start this work